FinanceManager.UnitOfWork/Abstructs/IUnitOfWork.cs

```diff
@@ -0,0 +1,13 @@
+using System.Data;
+
+namespace FinanceManager.UnitOfWork.Abstructs;
+
+public interface IUnitOfWork : IDisposable, IAsyncDisposable
+{
+    bool BulkOperation { get; set; }
+    T Repository<T>() where T : IUnitOfWorkRepository;
+    void Commit();
```

Comment:
Keep sync and async interface declarations separate to have ability to limit implementation to only required interface part (async only for example)